sotodlib/site_pipeline/preprocess_tod.py

```diff
@@ -152,12 +154,12 @@ def load_preprocess_det_select(obs_id, configs, context=None):
     """
     configs, context = _get_preprocess_context(configs, context)
     
-    pipe = Pipeline(configs["process_pipe"], logger=logger)
     meta = context.get_meta(obs_id)
-
-    for process in pipe:
-        logger.info(f"Selecting On {process.name}")
-        process.select(meta)
+    proc_aman = meta["preprocess"]
+    assn = [a for a in proc_aman._assignments][-1]
+    logger.info(f"Cutting on the last process: {assn}")
+    keep = core.flagman.has_all_cut(proc_aman[assn].valid)
+    meta.restrict('dets', meta.dets.vals[keep])
     return meta
 
 def load_preprocess_tod(obs_id, configs="preprocess_configs.yaml", context=None ):
```

Comment:
Currently this cannot load a single wafer, perhaps we want to update that but maybe not the purview of this PR.

Reply:
agree, beyond the scope of this PR